_pages/dev/session-management.md

```diff
@@ -0,0 +1,194 @@
+
+Authentication starts from the `AuthService` facade, where you initialize the process by calling one of the following login methods:
+
+- `loginWithCredentials` for the Resource Owner Password Flow
+- `loginWithRedirect` for the Implicit Flow or the Authorization Code Flow
+
+The login method then interacts with the `angular-oauth2-oidc` library. However, this interaction always goes through the `OAuthLibWrapperService`, which is a layer for isolating external libraries from Spartacus code.
+
+**Note:** If you want to access methods from a library that Spartacus does not expose, you should extend the `OAuthLibWrapperService` class.
+
+The `OAuthLibWrapperService` class is also responsible for configuring the library. It uses the `AuthConfigService` to access the `AuthConfig` configuration provided in the application.
+
+It is recommended that you always use the `AuthConfigService` if you need to read anything from the `AuthConfig`. This service is able to provide sensible defaults when some parts of the configuration are missing, and it gives you a few utilities out of the box, including one that detects OAuth flow based on your configuration.
+
+The `OAuthLibWrapperService` is a relatively thin layer that provides a lot of flexibility with the library, and also provides the `authentication.OAuthLibConfig` configuration option, which allows you to pass any configuration option to the library. Client and endpoint configurations are available in the top-level `authentication` configuration.
+
+You can learn more about advanced configuration of the authentication flow by looking at the `angular-oauth2-oidc` library source code, as well as the [angular-oauth2-oidc documentation](https://github.com/manfredsteyer/angular-oauth2-oidc).
+
+### Storing Tokens and User Identifiers
+
+After authentication, the tokens received from the library methods need to be stored somewhere. Previously, these tokens were kept in NgRx Store, but in Spartacus 3.0, there are dedicated services to keep the data. The library requires a storage mechanism with an API similar to `localStorage` or `sessionStorage`, and that was the main reason to switch from NgRx to services with a stream for keeping the data.
+
+![Storing auth data]({{ site.baseurl }}/assets/images/session-management/store.svg)
+
+For authentication, it is normally sufficient to store only tokens and their metadata (such as expiration time and scope). However, with OCC, there is also the tightly-coupled user ID that needs to be set after login or logout, as well as being required for user emulation when working with the Assisted Service Module (ASM). Prior to Spartacus 3.0, the user ID was kept in the same place as the tokens in NgRx, and because of that previous association, the user ID remains in the `UserAuthModule`. However, the tokens have now been separated from this module. Tokens and their metadata are now stored with the `AuthStorageService`, while user IDs have their own, dedicated `UserIdService`.
```

Comment:
```suggestion
For authentication, it is normally sufficient to store only tokens and their metadata (such as expiration time and scope). However, with OCC, there is also the tightly-coupled user ID that needs to be set after login or logout, as well as being required for user emulation when working with the Assisted Service Module (ASM). Prior to Spartacus 3.0, the user ID was kept in the same place as the tokens in NgRx, and because of that previous association, the user ID remains in the `UserAuthModule`. However, the tokens have now been separated from user identifier in this module. Tokens and their metadata are now stored with the `AuthStorageService`, while user IDs have their own, dedicated `UserIdService`.
```

Reply:
👍